animation_nodes/nodes/mesh/set_vertex_weights.py

```diff
@@ -0,0 +1,131 @@
+import bpy
+from bpy.props import *
+from ... events import propertyChanged
+from ... data_structures import VirtualDoubleList
+from ... base_types import AnimationNode, VectorizedSocket
+
+modeItems = [
+    ("ALL", "All", "Set weight of every vertex", "NONE", 0),
+    ("INDEX", "Index", "Set weight of a specific vertex", "NONE", 1)
+]
+
+groupIdentifierTypeItems = [
+    ("INDEX", "Index", "Get vertex group based on the index", "NONE", 0),
+    ("NAME", "Name", "Get vertex group based on the name", "NONE", 1)
+]
+
+groupNotFoundMessage = "group not found"
+noMeshMessage = "no mesh object"
+editMeshMessage = "object is not in object mode"
+
+class VertexWeights(bpy.types.Node, AnimationNode):
+    bl_idname = "an_SetVertexWeights"
+    bl_label = "Set Vertex Weights"
+    errorHandlingType = "EXCEPTION"
+
+    mode: EnumProperty(name = "Mode", default = "ALL",
+        items = modeItems, update = AnimationNode.refresh)
+
+    groupIdentifierType: EnumProperty(name = "Group Identifier Type", default = "INDEX",
+        items = groupIdentifierTypeItems, update = AnimationNode.refresh)
+
+    useIndexList: VectorizedSocket.newProperty()
+    useWeightsList: VectorizedSocket.newProperty()
+ 
+    def create(self):
+        self.newInput("Object", "Object", "object", defaultDrawType = "PROPERTY_ONLY")
+
+        if self.groupIdentifierType == "INDEX":
+            self.newInput("Integer", "Group Index", "groupIndex")
+        elif self.groupIdentifierType == "NAME":
+            self.newInput("Text", "Name", "groupName")
+
+        if self.mode == "INDEX":
+            self.newInput(VectorizedSocket("Integer", "useIndexList",
+                ("Index", "index"), ("Indices", "indices")))
+        self.newInput(VectorizedSocket("Float", "useWeightsList",
+                ("Weight", "weight"), ("Weights", "weights")))
+        
+        self.newOutput("Object", "Object", "object")
+
+    def draw(self, layout):
+        layout.prop(self, "mode", text = "")
+
+    def drawAdvanced(self, layout):
+        layout.prop(self, "groupIdentifierType", text = "Type")
+
+    def getExecutionFunctionName(self):
+        if self.mode == "INDEX":
+            if self.useIndexList:
+                if self.useWeightsList:
+                    return "execute_IndicesList"
```

Comment:
Perhaps using more explicit names would be better:

- `execute_IndicesList` -> `execute_Indices_WeightsList`
- `execute_IndicesSingle` -> `execute_Indices_SingleWeight`
- `execute_AllList` -> `execute_All_WeightsList`
- `execute_AllSingle` -> `execute_All_SingleWeights`

Reply:
I have renamed the "execute" functions.